.github/workflows/push.yml

```diff
@@ -57,6 +57,57 @@ jobs:
       - name: Publish test coverage
         uses: codecov/codecov-action@v4
 
+  tests-changes:
+    runs-on: ${{ matrix.os }}
+    continue-on-error: true
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - macos-latest
+          - ubuntu-latest
+          - windows-latest
+
+    steps:
+      - name: Checkout repository and submodules
```

Comment:
Create a reusable workflow for the first 5 steps since they are the same for both all and partial unit tests?
https://dev.to/n3wt0n/avoid-duplication-github-actions-reusable-workflows-3ae8

Reply:
The command to run can be passed as an arg to the common workflow.